### PR TITLE
Added BogoSort in Swift 4.1

### PR DESCRIPTION
**Pull request by CDsigma**
Added BogoSort in Swift 4.1

### CODE REVIEW

**Comment by Butt4cak3** [comment]
Welcome @CDsigma! Thanks for the contribution!

You're the first person to submit Swift code to the Algorithm Archive. You correctly added the code import to the .md file, but currently it doesn't know what `lang="swift"` means. For that, you have to go into *book.json* (which is at the root of this repository), scroll down to where all the programming languages are listed and add Swift like so:

```JSON
				...
				{
					"lang": "go",
					"name": "Go"
				},
				{
					"lang": "swift",
					"name": "Swift"
				}
```

I don't know if we have a Swift programmer to review your code, but we will see what we can do. Your code looks pretty good, so I'm sure it will get merged soon.

If you want, you can also add your name to the end of the *CONTRIBUTORS.md* file, but you don't have to.

**Comment by Butt4cak3** [comment]
Perfect. I totally didn't catch that you only changed `lang:"swift"` but not the other `"lang="swift"` attribute, but now it should be good. Now to wait for a Swift developer to show up. If we can't find one, I'll try to get your PR merged tomorrow or the day after.

**Comment by CDsigma** [comment]
Awesome :)

Thanks for your help

**Comment by Gustorn** [comment]
One small note: you commited with a git user/email different from your GitHub one. Was this intentional? If not, feel free to fix it and force push your changes.

File: chapters/sorting_searching/bogo/code/swift/bogoSort.swift [hunk 1]
```diff
@@ -0,0 +1,38 @@
+import Foundation
+
+
+func isSorted(inputArray: [Int]) -> Bool {
+
```
Comment (jiegillet) [inline chapters/sorting_searching/bogo/code/swift/bogoSort.swift]: Nitpick #1: 2 out of 3 functions start with an empty line. I don't know which style is best, but I do know that consistency is always rule #1 :)

File: chapters/sorting_searching/bogo/code/swift/bogoSort.swift [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+            return false
+        }
+    }
+
+    return true
+}
+
+
+
+func shuffle(inputArray: inout [Int]) -> [Int] {
```
Comment (jiegillet) [inline chapters/sorting_searching/bogo/code/swift/bogoSort.swift]: Nitpick 2: a quick Google search showed me that there is a native `shuffle` function in Swift 4.2+. Maybe it's best to use it.

Comment (Butt4cak3) [reply chapters/sorting_searching/bogo/code/swift/bogoSort.swift]: I wonder if there's a reason to use version 4.1 specifically.

Comment (jiegillet) [reply chapters/sorting_searching/bogo/code/swift/bogoSort.swift]: Oh my bad, I had not seen the version was specifically 4.1, it's right there in the title -_-
But yeah, I'd like to hear the reason too then :)

Comment (CDsigma) [reply chapters/sorting_searching/bogo/code/swift/bogoSort.swift]: Most people who use Swift tend to use the version which comes with Xcode. Currently the stable build of Xcode uses Swift 4.1 so I thought that would be the best choice. 

There is an Xcode 10 beta which uses Swift 4.2 but that won't be the stable build for a few months so until then I think Swift 4.1 is the best choice.

**Comment by CDsigma** [comment]
@Gustorn Ya I noticed that just after I committed. I didn't mean mean to do it but I also don't mind so I think I'll leave it 🤷

**Comment by Gustorn** [comment]
@CDsigma sure thing, I just thought I'd mention it since sometimes people accidentally commit with their work email (did it once or twice myself) and they'd rather not have that online. If you're OK with the current one it's all good.

**Comment by Butt4cak3** [comment]
The squashed commit will have the right author, I think. It won't really matter in the end.

File: book.json [hunk 3]
```diff
@@ -84,6 +84,10 @@
 					"name": "Go"
 				},
 				{
+					"lang": "swift",
+					"name": "Swift"
+     				},
+      				{
```
Comment (Butt4cak3) [inline book.json]: The two lines with the brackets are indented with a bunch of spaces and then 3 tabs. The file is tab indented. Try to make sure you only use tabs.

File: chapters/sorting_searching/bogo/bogo_sort.md [hunk 4]
```diff
@@ -36,6 +36,9 @@ In code, it looks something like this:
 {% sample lang="rs" %}
 [import, lang:"rust"](code/rust/bogosort.rs)
 {% endmethod %}
+{% sample lang="swift" %}
+[import, lang:"swift"](code/swift/bogoSort.swift)
```
Comment (Butt4cak3) [inline chapters/sorting_searching/bogo/bogo_sort.md]: I don't like the capitalization of the "S" in the file name. Some languages (like Java) require capitalization to match classes and such inside the files, but I'd prefer to keep everything else lowercase.

Unless it's some kind of convention in Swift. In that case, keep it.

File: book.json (hunk 3 above)
Comment (CDsigma) [reply book.json]: Sorry about that just committed some changes so no more spaces are used

File: chapters/sorting_searching/bogo/bogo_sort.md (hunk 4 above)
Comment (CDsigma) [reply chapters/sorting_searching/bogo/bogo_sort.md]: Just updated the file name 👍

**Comment by Butt4cak3** [comment]
Since we all don't really *know* Swift, I'll just take the liberty to merge this one. The logic of the code is correct and so it shouldn't be an issue.

**Comment by Butt4cak3** [comment]
Nevermind. I just found a mistake that breaks the build. You have to fix this before I can merge.

File: chapters/sorting_searching/bogo/bogo_sort.md [hunk 5]
```diff
@@ -36,6 +36,9 @@ In code, it looks something like this:
 {% sample lang="rs" %}
 [import, lang:"rust"](code/rust/bogosort.rs)
 {% endmethod %}
+{% sample lang="swift" %}
+[import, lang:"swift"](code/swift/bogosort.swift)
+{% endmethod %}
```
Comment (Butt4cak3) [inline chapters/sorting_searching/bogo/bogo_sort.md]: There's only supposed to be one `{% endmethod %}` statement at the end. You have to remove the one in line 38.

**Pull request merged**